x-pack/filebeat/input/googlepubsub/input.go

```diff
@@ -152,15 +160,11 @@ func (in *pubsubInput) run() error {
 	// Start receiving messages.
 	topicID := makeTopicID(in.ProjectID, in.Topic)
 	return sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
```

Comment:
Is there any impact of letting the `func` return without and `Ack/Nack()` (is there an impact of not handling the message synchronously in `f`)? I wonder if it affects the accounting used to track MaxOutstandingMessages/MaxOutstandingBytes. Or if it affect the automatic ack deadline extension of the message. The docs for `Receive` aren't 100% clear to me.

Reply:
Yes, the docs aren't too clear. My understanding is that it will only affect the ACK deadline if it gets ACK'd at a later time (before re-transmission).

In my tests (against gcloud, not the emulator) a non-ACKed message gets re-delivered eventually after a few minutes. Nack() just causes it to redeliver more quickly.